### PR TITLE
[Streaming] refine streaming tests sleep condition

### PR DESCRIPTION
**Pull request by chaokunyang**
## Why are these changes needed?
Refine streaming tests sleep condition

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ashione — APPROVED** [review]

File: streaming/python/tests/test_word_count.py [hunk 1]
```diff
@@ -45,14 +46,23 @@ def sink_func(x):
...
-    import time
-    time.sleep(3)
+
+    def check_succeed():
+        if os.path.exists(sink_file):
+            import time
+            time.sleep(3)  # Wait all data be written
+            with open(sink_file, "r") as f:
+                result = f.read()
+                assert "a:2" in result
```
Comment (kfstorm) [inline streaming/python/tests/test_word_count.py:56]: If the assertion fails, the exception will be propagated and the test case will still fail.

File: streaming/python/tests/test_word_count.py [hunk 2]
```diff
@@ -45,14 +46,23 @@ def sink_func(x):
                 (old_value[0], old_value[1] + new_value[1])) \
         .sink(sink_func)
     ctx.submit("word_count")
-    import time
-    time.sleep(3)
+
+    def check_succeed():
+        if os.path.exists(sink_file):
+            import time
+            time.sleep(3)  # Wait all data be written
```
Comment (kfstorm) [inline streaming/python/tests/test_word_count.py:53]: We don't need this anymore.

File: streaming/python/tests/test_word_count.py [hunk 3]
```diff
@@ -45,14 +46,23 @@ def sink_func(x):
...
+            with open(sink_file, "r") as f:
+                result = f.read()
+                assert "a:2" in result
+                assert "b:2" in result
+                assert "c:2" in result
+            print("Execution succeed")
```
Comment (kfstorm) [inline streaming/python/tests/test_word_count.py:59]: Remove this line

File: streaming/python/tests/test_word_count.py (hunk 1 above)
Comment (chaokunyang) [reply streaming/python/tests/test_word_count.py:56]: If the assertion fails, it indicates some srrors occored, the test should fail in that case

File: streaming/python/tests/test_word_count.py (hunk 2 above)
Comment (chaokunyang) [reply streaming/python/tests/test_word_count.py:53]: We write file using append mode, so we need to wait a little for all data being written

File: streaming/python/tests/test_word_count.py (hunk 3 above)
Comment (chaokunyang) [reply streaming/python/tests/test_word_count.py]: removed

File: streaming/python/tests/test_word_count.py (hunk 1 above)
Comment (kfstorm) [reply streaming/python/tests/test_word_count.py:56]: What if the file is just created and the content is to be written?

Comment (chaokunyang) [reply streaming/python/tests/test_word_count.py:56]: Tha't why we wait 3 seconds

File: streaming/python/tests/test_word_count.py (hunk 2 above)
Comment (kfstorm) [reply streaming/python/tests/test_word_count.py:53]: You basically only need to provide a function which simply returns true or false and let `wait_for_condition` handle the sleep and retry.

Comment (chaokunyang) [reply streaming/python/tests/test_word_count.py]: fixed

File: streaming/python/tests/test_word_count.py [hunk 4]
```diff
@@ -45,14 +46,17 @@ def sink_func(x):
                 (old_value[0], old_value[1] + new_value[1])) \
         .sink(sink_func)
     ctx.submit("word_count")
-    import time
-    time.sleep(3)
+
+    def check_succeed():
+        if os.path.exists(sink_file):
+            with open(sink_file, "r") as f:
```
Comment (kfstorm) [inline streaming/python/tests/test_word_count.py:52]: Do we need to deal with IO error here?

Comment (chaokunyang) [reply streaming/python/tests/test_word_count.py:52]: If file exists, open and file shouldn't fail.

**Review by kfstorm — APPROVED** [review]

**Pull request merged**